modules/samples/firealarm/component/manager/src/main/java/org/homeautomation/firealarm/manager/api/ManagerService.java

```diff
@@ -0,0 +1,337 @@
+            }
+            return added;
+        } catch (DeviceManagementException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error(e.getErrorMessage(), e);
+            return false;
+        } catch (AccessTokenException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+            log.error("Unable to obtain access token", e);
+            return false;
+        }
+    }
+
+    /**
+     * Remove installed device
+     *
+     * @param deviceId unique identifier for device
+     * @param response to request
+     */
+    @Path("manager/device/remove/{device_id}")
+    @DELETE
+    public void removeDevice(@PathParam("device_id") String deviceId,
+                             @Context HttpServletResponse response) {
+        DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+        deviceIdentifier.setId(deviceId);
+        deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+        try {
+            boolean removed = APIUtil.getDeviceManagementService().disenrollDevice(
+                    deviceIdentifier);
+            if (removed) {
+                response.setStatus(Response.Status.OK.getStatusCode());
+            } else {
+                response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+        } catch (DeviceManagementException e) {
+            response.setStatus(Response.Status.INTERNAL_SERVER_ERROR.getStatusCode());
+        }
+    }
+
+    /**
+     * Update device name
+     *
+     * @param deviceId unique identifier for device
+     * @param name     new name of the device
+     * @param response to request
+     * @return update status
+     */
+    @Path("manager/device/update/{device_id}")
+    @POST
+    public boolean updateDevice(@PathParam("device_id") String deviceId,
+                                @QueryParam("name") String name,
+                                @Context HttpServletResponse response) {
+        DeviceIdentifier deviceIdentifier = new DeviceIdentifier();
+        deviceIdentifier.setId(deviceId);
+        deviceIdentifier.setType(DeviceTypeConstants.DEVICE_TYPE);
+        try {
+            Device device = APIUtil.getDeviceManagementService().getDevice(deviceIdentifier);
+            device.setDeviceIdentifier(deviceId);
+            device.getEnrolmentInfo().setDateOfLastUpdate(new Date().getTime());
+            device.setName(name);
+            device.setType(DeviceTypeConstants.DEVICE_TYPE);
+            boolean updated = APIUtil.getDeviceManagementService().modifyEnrollment(device);
+            if (updated) {
+                response.setStatus(Response.Status.OK.getStatusCode());
+            } else {
+                response.setStatus(Response.Status.NOT_ACCEPTABLE.getStatusCode());
+            }
+            return updated;
+        } catch (DeviceManagementException e) {
+            log.error(e.getErrorMessage());
```

Comment:
When logging an error, it should be something custom and specific to API layer as the exception actually occurs due to a call at that level. And also, it's better to pass the error stack-trace to be logged, so that errors and it's causes could be traced so easily at the event of an exception. This can be done simply by passing the exception object being thrown, to log.error function as parameter.

log.error(errorMessage, e);